sensorprocess/sensorprocess.go

```diff
@@ -10,34 +10,52 @@ import (
 
 	"github.com/edaniels/golog"
 	"go.viam.com/rdk/components/camera/replaypcd"
+	replaymovementsensor "go.viam.com/rdk/components/movementsensor/replay"
 
 	"github.com/viamrobotics/viam-cartographer/cartofacade"
 	"github.com/viamrobotics/viam-cartographer/sensors"
 )
 
+var undefinedIMU = cartofacade.IMUReading{}
+
 // Config holds config needed throughout the process of adding a sensor reading to the cartofacade.
 type Config struct {
 	CartoFacade              cartofacade.Interface
 	Lidar                    sensors.TimedLidarSensor
 	LidarName                string
 	LidarDataRateMsec        int
+	IMU                      sensors.TimedIMUSensor
+	IMUName                  string
+	IMUDataRateMsec          int
 	Timeout                  time.Duration
 	Logger                   golog.Logger
+	nextData                 nextData
+	started                  bool
+	stopped                  bool
 	RunFinalOptimizationFunc func(context.Context, time.Duration) error
 }
 
-// Start polls the lidar to get the next sensor reading and adds it to the cartofacade.
+// nextData stores the next data to be added to cartographer along with its associated timestamp so that,
+// in offline mode, data from multiple sensors can be added in order.
+type nextData struct {
+	lidarTime time.Time
+	lidarData []byte
+	imuTime   time.Time
+	imuData   cartofacade.IMUReading
+}
+
+// StartLidar polls the lidar to get the next sensor reading and adds it to the cartofacade.
 // stops when the context is Done.
-func Start(
+func (config *Config) StartLidar(
 	ctx context.Context,
-	config Config,
 ) bool {
```

Comment:
what do you think of adding this chunk of code here so that we only set started once? Not sure if its worth making this function clunkier but because it happens everytime we add a lidar reading when I first read it I though we were setting this back and forth

```
if config.LidarataRateMsec == 0 {
    tryAddLidarReadingUntilSuccess(ctx, tsr.Reading, tsr.ReadingTime, config)
    config.started = true
}
```

Reply:
Spoke offline, we are moving towards using a start and end time instead of bools to more clearly describe whats happening